yt_dlp/extractor/espn.py

```diff
@@ -279,3 +278,133 @@ def _real_extract(self, url):
             'formats': formats,
             'subtitles': subtitles,
         }
+
+
+class WatchESPNIE(AdobePassIE):
+    _VALID_URL = r'https://www.espn.com/watch/player/_/id/(?P<id>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})'
+    _TESTS = [{
+        'url': 'https://www.espn.com/watch/player/_/id/6eee6cb0-1795-49e3-84ea-766b517e0309',
+        'info_dict': {
+            'id': '6eee6cb0-1795-49e3-84ea-766b517e0309',
+            'ext': 'mp4',
+            'title': 'Miami vs. #2 Duke',
+            'thumbnail': 'https://artwork.api.espn.com/artwork/collections/media/6eee6cb0-1795-49e3-84ea-766b517e0309/default?width=640&apikey=1ngjw23osgcis1i1vbj96lmfqs',
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }, {
+        'url': 'https://www.espn.com/watch/player/_/id/bd1f3d12-0654-47d9-852e-71b85ea695c7',
+        'info_dict': {
+            'id': 'bd1f3d12-0654-47d9-852e-71b85ea695c7',
+            'ext': 'mp4',
+            'title': 'Tue, 1/11 - ESPN FC',
+            'thumbnail': 'https://s.secure.espncdn.com/stitcher/artwork/collections/media/bd1f3d12-0654-47d9-852e-71b85ea695c7/16x9.jpg?timestamp=202201112217&showBadge=true&cb=12&package=ESPN_PLUS',
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }]
+
+    _API_KEY = 'ZXNwbiZicm93c2VyJjEuMC4w.ptUt7QxsteaRruuPmGZFaJByOoqKvDP2a5YkInHrc7c'
+
+    def _call_bamgrid_api(self, path, video_id, data=None, headers={}):
+        if 'authorization' not in headers:
+            headers['authorization'] = 'Bearer ' + self._API_KEY
+        return self._download_json(
+            'https://espn.api.edge.bamgrid.com/' + path,
+            video_id, data=data, headers=headers)
+
+    def _payload_to_string_data(self, payload):
+        return '&'.join(['%s=%s' % (key, compat_urllib_parse_quote_plus(payload[key])) for key in payload]).encode('utf-8')
+
+    def _payload_to_json_data(self, payload):
+        return json.dumps(payload).encode('utf-8')
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        video_data = self._download_json(
+            'https://watch-cdn.product.api.espn.com/api/product/v3/watchespn/web/playback/event?id=%s' % video_id, video_id)['playbackState']
+        title = video_data['name']
+
+        # ESPN+ subscription required, through cookies
+        if video_data.get('sourceId') == 'ESPN_DTC':
+            cookies = self._get_cookies(url)
+            try:
+                id_token = cookies['ESPN-ONESITE.WEB-PROD.token'].value.split('|')[1]
+            except KeyError:
+                raise ExtractorError('This is an ESPN+ video, which requires cookies. Use --cookies or --cookiesfrombrowser')
+
```

Comment:
1. check with .get instead of KeyError
2. self.raise_login_required

Reply:
Should be fixed